lib/datasource.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const debug = require('debug')('graphql-component:dataSource');
+
+const intercept = function (instance, context) {
+  debug(`intercepting ${instance.constructor.name}`);
+
+  return new Proxy(instance, {
+    get(target, key) {
+      if (typeof target[key] !== 'function' || key === instance.constructor.name) {
+        return target[key];
+      }
+      const original = target[key];
+      
+      return function (...args) {
+        return original(context, ...args);
```

Comment:
very cool - had to do a quick read on how `Proxy` works but I see this is how you are getting context injected as the first argument of every function in a data source class.